src/CMakeLists.txt

```
@@ -81,6 +81,8 @@ if(BUILD_SHARED_LIBS)
   set (ONEMKL_LIBS ${TARGET_DOMAINS})
   list(TRANSFORM ONEMKL_LIBS PREPEND onemkl_)
   target_link_libraries(onemkl PUBLIC ${ONEMKL_LIBS})
+  set_target_properties(onemkl PROPERTIES
+      INSTALL_RPATH "\$ORIGIN")
```

```suggestion
  set_target_properties(onemkl PROPERTIES
      INSTALL_RPATH "\$ORIGIN"
      BUILD_RPATH "\$ORIGIN"
  )
```

I was just testing this. `BUILD_RPATH` is needed when running the examples without running the install step.

---

Nice catch! But I think `BUILD_WITH_INSTALL_RPATH TRUE` would be a bit better, since CMake (at least 3.30) is not smart enough to see that `INSTALL_RPATH` and `BUILD_RPATH` are the same, and still tries to modify the SO when installing. Quite fast and harmless, but still.